app/helpers/grade_entry_forms_helper.rb

```diff
@@ -29,7 +29,7 @@ def update_grade_entry_form_params(attributes)
 
     unless grade_entry_items.nil?
       # Update the attributes hash
-      max_position = 1
+      max_position = GradeEntryItem.where(:grade_entry_form_id => 1).count
```

Comment:
This is the right idea, but this hard-codes a grade entry form id of `1`, which isn't good. You should be able to get the correct id from `attributes`.

Reply:
Actually I'm looking at the code immediately beneath this now, which does seem to correctly increment `max_position` for each grade entry item. The way the form is set up, both existing and new grade entry items should appear in `params[:grade_entry_form][:grade_entry_items_attributes]`, so there's a mixture of both items with existing positions, and ones that are new.

I wonder if what's actually causing the problem is that the items aren't sorted in `grade_entry_items`, so they're assigned random new positions. You could try fixing this by explicitly sorting `grade_entry_items` by `position`, putting the ones with `nil` last.